File: 14281.kt
/*
# https://www.acmicpc.net/problem/14281

from sys import stdin

input = stdin.readline

n = int(input())
seq = list(map(int, input().rstrip().split()))

if len(seq) <= 2:
    print(0)
else:
    copy = seq.copy()
    ii = 1
    while ii < n - 1:
        if copy[ii] * 2 > copy[ii - 1] + copy[ii + 1]:
            copy[ii] = (copy[ii - 1] + copy[ii + 1]) // 2
            ii = ii - 1 if ii != 1 else 1
        else:
            ii += 1

    # print(copy)
    print(sum(seq) - sum(copy))
    

 */
// this python code to kotlin..

File: problems/14281.py
# https://www.acmicpc.net/problem/14281

from sys import stdin

input = stdin.readline

n = int(input())
arr = list(map(int, input().rstrip().split()))

if len(arr) <= 2:
    print(0)
else:
    seq = arr.copy()
    while True:
        convex = True
        
        for ii in range(1, n-1):
            if seq[ii] * 2 > seq[ii - 1] + seq[ii + 1]:
                seq[ii] = (seq[ii - 1] + seq[ii + 1]) // 2
                convex = False
        
        if convex:
            break
        
    print(sum(arr) - sum(seq))
